src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -464,7 +464,9 @@ private async Task OnPartialValidationSucceededAsync(ChainedHeader chainedHeader
                 }
             }
 
-            if (chainedHeaderBlocksToValidate != null)
+            bool nextValidationNeeded = chainedHeaderBlocksToValidate != null;
+            bool fullValidationFailed = (connectBlocksResult != null) && !connectBlocksResult.Succeeded;
```

Comment:
I'd add a comment saying that in case `connectBlocksResult` is null it means that full validation didn't happen

Reply:
Will add a comment 👍